common/openapi.yml

```
@@ -297,6 +321,36 @@ components:
       description: >
         DatasetPreviewTablesResponse returns information about the table structure of a dataset.
 
+    DatasetPreviewTable:
+      type: object
+      properties:
+        columns:
+          type: array
+          items:
+            $ref: '#/components/schemas/ColumnModel'
+        rows:
+          type: array
+          items:
+            type: object
+
+    ColumnModel:
+      type: object
+      properties:
+        name:
+          type: string
+        datatype:
+          $ref: '#/components/schemas/TableDataType'
+        array_of:
+          type: boolean
+          description: if true, make this column an array of type datatype.
```

This sounds like the description for creation, not for an API result 
```suggestion
          description: if true, this column is an array of data
```